### ISSUE TITLE
Composer runtime (0.19.6) is not compatible with client (0.20.0) - Multi org fabric 1.2 issue

### ISSUE THREAD

**Issue by abilashiim** [hyperledger/composer#4335]
Not able to do a Composer network ping or complete the process of deploying a multi-org Hyperledger Fabric 1.2 using the latest docs

## Context
          Setting up a multi-org Hyperledger Fabric 1.2 using the latest tutorial.
          Deployment fails at Composer network ping
  

## Expected Behavior
         Successful network ping an business card FQDN for accessing business network

## Actual Behavior
 abilash@HyperledgerNew:/tmp/composer/org1$ composer network ping -c abilash@tutorial-network
Error: Error trying to ping. Error: Composer runtime (0.19.6) is not compatible with client (0.20.0)
Command failed

## Possible Fix
May be some problem with software bundling of the new Fabric 1.2.

## Steps to Reproduce
Just follow the latest Fabric 1.2 composer -  Multi org tutorial.
https://hyperledger.github.io/composer/latest/tutorials/deploy-to-fabric-multi-org.html

## Existing issues
I tried upgrading composer-rest-server, composer-cli etc as suggested in previous bugs, but no respite. Even made a completely new deployment. Same issue.

## Context
Not able to get the Multi org setup with Fabric 1.2 up and running.

## Your Environment
* Version used:  Fabric 1.2
* Environment name and version (e.g. Chrome 39, node.js 5.4): Azure
* Operating System and version (desktop or mobile): Ubuntu 16.0.4

**Comment by davidkel** [comment]
There are 2 ways you could have deployed a bna file that references the composer-runtime at version 0.19.6. The first is that you were using composer-cli or composer-playground whose version was 0.19.6. The second is that the bna file you are deploying has an explicit reference to composer-runtime. 
You can check your bna file by using an unzip tool to unpack it and check the package.json for dependencies. Alternatively try creating a new bna file from scratch (ie from the source of a business network where you can inspect the package.json beforehand to ensure there are no references to composer-* modules) and make sure you use composer-cli at version 0.20.0 to install that newly created package.

**Comment by wompeter** [comment]
I've wrestled with this issue all day.  I was also following the documentation and ran into this error.  I have previously done hyperledger tutorials with 0.19.13.

I nuked all existing docker containers, fabric-dev-servers, docker images.

Despite all package.json files in my bna project AND composer runtime installs saying they were version 0.20.0 (including composer -V reporting itself as 0.20.0), I still received the error "runtime (0.19.13) not compatible with client (0.20.0)".   Very frustrating.

The only thing that solved it was going into my global npm cache and physically deleting all composer libraries before re-issuing npm i -g commands again.

**Comment by ronaldlong46** [comment]
Another thing to look out for. Check your package.json in your BNA file for something like this:

`{"composer-common":"0.19.13","composer-runtime-hlfv1":"0.19.13"}`

It should be:
`{"composer-common":"0.20.0","composer-runtime-hlfv1":"0.20.0"}`

That was the cause of my issues.

**Comment by abilashiim** [comment]
Yes the issue was with the .bna file having 0.19.6 as version number. 

For serviceability, I would recommend that while we install the business network, it should throw an error saying "incompatible version" or composer should be backward compatible with the older version .bna files.

Current approach of allowing older .bna files to be imported and throwing error at a later stage does not make sense or adds to usability. Request Hyperledger team to improve deployment on this context.

**Comment by raymondfx** [comment]
Where can I find .bna file, seems I cannot find it. I have the same problem

**Comment by abhishekover9000** [comment]
I was having this same problem and found this. @raymondfx the bna file is generated from archive command.

**Comment by anshulgupta-1985** [comment]
> There is a further way this problem could have occurred. You may still have a chaincode docker image of your business network created using composer 0.19 and fabric 1.1. To ensure you run cleanly you should delete all docker images of chaincode images. You can either delete all your docker images and start again using
> 
> ```
> docker rmi $(docker images -q)
> ```
> 
> or if you are using byfn or the fabric-dev-servers package then you should be able to delete all the chaincode docker images as follows
> 
> ```
> docker rmi $(docker images dev-* -q)
> ```
> 
> Just make sure running docker containers are killed and removed first
> 
> ```
> docker kill $(docker ps -q)
> docker rm $(docker ps -aq)
> ```

Hi I also faced the same issue Error: Error trying to ping. Error: Composer runtime (0.19.6) is not compatible with client (0.20.0)
Command failed

But now issue is getting resolved. What i did use this link https://hyperledger.github.io/composer/latest/installing/development-tools

and use the following commands to destroy the previous installations
    docker kill $(docker ps -q)
    docker rm $(docker ps -aq)
    docker rmi $(docker images dev-* -q)

Uninstalling the development environment 
https://hyperledger.github.io/composer/latest/installing/uninstall-dev-env

Installaing the development environment

![image](https://user-images.githubusercontent.com/51150816/58535117-b46f7d00-820a-11e9-9dbf-38fe7ee03851.png)

then go with the development of the business network creation as it is mentioned steps.
